src/main/java/org/folio/service/invoice/POLInvoiceLineRelationService.java

```diff
@@ -112,32 +112,41 @@ private static org.folio.rest.acq.model.invoice.FundDistribution convertToInvoic
 
   private void copyAmountsAndRecalculateNewEncumbrance(List<Transaction> forCreate, List<Transaction> forDelete, List<InvoiceLine> invoiceLines, String currency) {
     // Update encumbrances when an order line is linked to a paid invoice
-    double amountExpended = forDelete.stream().map(encumbrance -> encumbrance.getEncumbrance().getAmountExpended())
-      .map(BigDecimal::valueOf).reduce(BigDecimal.ZERO, BigDecimal::add).doubleValue();
-    double amountAwaitingPayment = forDelete.stream().map(encumbrance -> encumbrance.getEncumbrance().getAmountAwaitingPayment())
-      .map(BigDecimal::valueOf).reduce(BigDecimal.ZERO, BigDecimal::add).doubleValue();
+    double amountExpended = FinanceUtils.sumAmounts(forDelete, Encumbrance::getAmountExpended);
+    double amountCredited = FinanceUtils.sumAmounts(forDelete, Encumbrance::getAmountCredited);
+    double amountAwaitingPayment = FinanceUtils.sumAmounts(forDelete, Encumbrance::getAmountAwaitingPayment);
+
     boolean isReleaseEncumbranceEnabled = filterInvoiceLinesByStatuses(invoiceLines, List.of(InvoiceLine.InvoiceLineStatus.PAID))
       .stream()
       .anyMatch(InvoiceLine::getReleaseEncumbrance);
 
     forCreate.stream().findFirst().ifPresent(transaction -> {
       Encumbrance encumbrance = transaction.getEncumbrance();
-      double encumbranceAmount = HelperUtils.calculateEncumbranceEffectiveAmount(encumbrance.getInitialAmountEncumbered(),
-        amountExpended, amountAwaitingPayment, Monetary.getCurrency(currency));
+      double encumbranceAmount = FinanceUtils.calculateEncumbranceEffectiveAmount(encumbrance.getInitialAmountEncumbered(),
+        amountExpended - amountCredited, amountAwaitingPayment, Monetary.getCurrency(currency));
```

Comment:
can you optimize this. for example if possible move minus calculation to the inside of method

Reply:
Unfortunately we can't. `FundsDistributionService` uses same calculation but with different *initialAmount*

Reply:
I agree,  I expect that this name getExpended() return only expended amount, but not expended - credited. So need to rename in some way